irods_testing_environment/irods_setup.py

```diff
@@ -350,8 +350,29 @@ def restart_rsyslog(container):
         if ec != 0:
             logging.info(f'[{container.name}] failed to kill rsyslogd')
 
-        ec = execute.execute_command(container, rsyslog_bin_path)
-        if ec != 0:
+        # TODO: Remove multiple attempts when a more appropriate solution is found
+        MAX_NUMBER_OF_ATTEMPTS = 3
+        num_attempts = 1
+
+        logging.debug("Attempting startup of rsyslogd")
+
+        while num_attempts <= MAX_NUMBER_OF_ATTEMPTS:
+            logging.debug("Running startup attempt [#{}]".format(num_attempts))
+            ec = execute.execute_command(container, rsyslog_bin_path)
+            logging.debug("Startup attempt [#{}] {status}.".format(num_attempts, status="succeeded" if ec == 0 else "failed"))
+
+            logging.debug("Checking to see if rsyslogd started up in the background.")
+            is_alive = execute.execute_command(container, f'pgrep {os.path.basename(rsyslog_bin_path)}') == 0
+            logging.debug("Result of checking if rsyslogd is running: [{}]".format(is_alive))
```

Comment:
Similar to the `RuntimeError` message below, might consider adding the `container.name` to each of these log messages since multiple iRODS-es can be starting up failing/succeeding at once :)

Reply:
yes please